spec/ics-027-direct-interchain-exchange/ReadMe.md

```diff
@@ -0,0 +1,74 @@
+Bob perceive the security of Foo Chain to be /moderate/.
+Or, Bob perceives the security of both FooChain and BarChain to be *strong*.
+
+#### Protocol
+
+For simplicity, we will describe only one scenario, where Alice wishes to submit a limit order to purchase Bar tokens with her Foo tokens, where the desired price is lower than market (e.g. the limit order will not be matched immediately).
+
+Alice submits to FooChain a limit order transaction on FooChain: {type:"limit", pair:"bar/foo", sell:"20foo", price:"0.9"}
+
+FooChain sees that the price of "0.9" is lower than the current price of "1", so it locks Alice's 20 Foo tokens, and sends this order to BarChain via IBC.
+
+This order persists on FooChain (in the form of locked tokens) and BarChain (in the form of an open limit order) until one of two things happen.
+
+Case A: Alice cancels the order by submitting a cancellation order on BarChain: {type:"limit-cancel", pair:"bar/foo", sell:"20foo", account:"Alice"}.
+Case B: Bob (say) partially or fully matches the order by submitting a limit order on BarChain: {type:"limit", pair:"bar/foo", sell:"100bar", price:"0.8"}.
+
+There are ways to design the protocol such that Alice can cancel the order without having to submit an order to BarChain, but due to the asynchrony of IBC it cannot be immediate for both Alice and Bob simultaneously.  Such extensions to the interchain exchange protocol are not described here.
+
+For both case A and case B, a corresponding opposite IBC message is created from BarChain to FooChain.  In case A, the 20 Foo tokens are released back to Alice.  In case B, the 20 Foo tokens are released back to Bob, and in this example (since Bob's order volume was greater than Alice's), a new partial limit order would be created on FooChain on behalf of Bob.
+
+#### Delivery
+
+While the delivery of matched orders may be routed anywhere, by default we propose that the delivery happens on the Hub as IBC pegged tokens.
+It simplifies custody for Alice and Bob because of common custodial protocols on the Hub.
+It creates a substantial social pressure toward the economic finality of the exchange of tokens by having them delivered to Hub accounts.
+Only delivered tokens need to be on the Hub, not the tokens behind pending limit orders, so as long as users move their funds out of the Hub soon after execution, the security requirement on the hub is minimal.
+
+#### Hub Collateralization
+
+With N^2 IBC connections between N zones for this Direct Interchain Exchange Protocol (DIEP), it becomes difficult for traders and market makers to make sense of the complexity arising from chain security.
+Additionally, by forcing a chain to fail with a double spend attack (a consensus fork), it may become profitable to induce failure, especially with more complex scenarios that require consistency across the interchain.  For example, interchain pair trading may become profitable to attack.
```

Comment:
I'd like to see a bullet point example of how a double spend attack would happen (and affect the hub-collateral) in this cross-chain context.

Reply:
I think the writer is explaining that there might exist high probability of IBC malfunctioning when asset control is too decentralized into so many zones. 

if there exists a bullet point example, it means there exists a bug in IBC. Point is that we will have more risk to experience IBC bug in more complicated design.